### PR TITLE
Provide simple Gaussian broadening

### PR DESCRIPTION
**Pull request by ajjackson**
The implementation uses convolution on a regular energy grid; this is more efficient than evaluating the Guassian at every point, but invalid for irregular energy grids. I have included a check for this condition, but due to the limited precision of input file data there is quite a generous tolerance of 1e-5.

There is a UI/API decision to be made around how we represent "no broadening". I think the cleanest options are:

- default to `None`, which represents no broadening. Behaviour of sigma <= 0. is undefined.
- default to some recommended broadening value. User can disable broadening by setting sigma <= 0.
- default to some recommended broadening, and provide another `--no-broadening` option that sets sigma to `None` and disables it.

The choice of whether to broaden by default should be science-led. If so, the different representations of no broadening are a trade-off in convenience vs clarity.

- [x] Implement broadening option to plotter
- [x] Provide CLI access to this option for auto plots and manual plots
- [ ] Add tests
- [ ] Document with example

### CODE REVIEW

**Comment by ajjackson** [comment]
It looks like the test failures are cause by a Pymatgen update, reducing verbosity of the output files. I guess we need to update the tests?

**Comment by JaGeo** [comment]
Yes, we need to update the tests. I reduced the verbosity in one of the last versions of pymatgen. (Sorry, this was on my todo for one the next weeks ...)

I personally think that no broadening as a default would be the best option. I am not sure users would expect a broadening for COHPs/COBIs/COOPs. Otherwise, we would need to print something to the screen that broadening is added.

**Comment by JaGeo** [comment]
I wanted to fix another issue anyway. I will start fixing this test as well.

**Comment by JaGeo** [comment]
Tests are passing now

**Comment by ajjackson** [comment]
I'm really confused by this test failure. Obviously it's something to do with the stdout citation boilerplate, but... it works on my machine and seems to be consistent with the non-broadened entries?

**Comment by JaGeo** [comment]
Thanks! Will try to have a look at it 🙂

**Comment by JaGeo** [comment]
@ajjackson It passes :). I recreated the json and everything worked for me. Maybe a problem with the pymatgen version?

Is there anything else that would need to be done before I could merge? I will also shortly test it now.

**Comment by JaGeo** [comment]
@ajjackson, could you check (when you have some time) if this last change resolves the issue with the broadening that we talked about. Based on this example, I thought the normalization might be missing: https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.convolve.html 

Happy to check in more detail in one of the next days as well.

File: lobsterpy/plotting/__init__.py [hunk 1]
```diff
@@ -219,4 +219,5 @@ def _broaden(energies: np.ndarray, population: np.ndarray, sigma=None, cutoff=4.
         kernel_x = np.concatenate([-kernel_x[-1:1:-1], kernel_x])
 
         kernel = norm.pdf(kernel_x, scale=sigma)
-        return convolve(population, kernel, mode="same")
+
+        return convolve(population, kernel, mode="same") / sum(kernel)
```
Comment (ajjackson) [inline lobsterpy/plotting/__init__.py:223]: That'll do it, cheers!

`kernel.sum()` is much faster than the generic `sum`, but it's microseconds so not really a big deal.

Comment (JaGeo) [reply lobsterpy/plotting/__init__.py:223]: Great. Will make this change as well and then use this for plotting. Thanks!

**Pull request merged**